src/blockchain_db/blockchain_db.cpp

```diff
@@ -423,17 +423,15 @@ void BlockchainDB::trim_block()
 
   // Trim the tree to the new n leaf tuples
   this->trim_tree(new_n_leaf_tuples, tree_block_idx);
+
+  // Remove block from tree meta
+  this->del_tree_meta(tree_block_idx);
```

Comment:
Is `trim_block()` the function this call is supposed to be in or `trim_layers()`? B/c I think it changed

Reply:
`trim_block` is good. Notice it was originally in `trim_tree`, which is only called by `trim_block`.

If it was in `trim_layers` and there are no layers to trim, then the tree meta for the block wouldn't get deleted. It should get deleted when removing the block.